File: src/model/CoachingPlan.js
export default class CoachingPlan {

    constructor(source) {
        if (source === undefined) {
            this.goal = '';
            this.principles = [];
            this.priorities = [];
            this.actions = [];
            this.startDate = new Date();
            this.reviewDate = null;
            this.notes = '';
            this.updatedAt = null;
            return;
        }
        this.goal = source.goal;
        this.principles = source.principles;
        this.priorities = source.priorities;
        this.actions = source.actions;
        this.startDate = toDate(source.startDate);
        this.reviewDate = source.reviewDate ? toDate(source.reviewDate) : null;
        this.notes = source.notes || '';
        this.updatedAt = source.updatedAt;
    }
}

function toDate(value) {
    return value instanceof Date ? value : new Date(`${value}T12:00:00`);
}

File: backend/src/main/java/com/jllado/weightcontrol/domain/CoachingPlan.java
package com.jllado.weightcontrol.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Convert;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import java.time.Instant;
import java.time.LocalDate;
import java.util.List;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "coaching_plans", uniqueConstraints = @UniqueConstraint(name = "uq_coaching_plans_user", columnNames = "user_id"))
@Getter
@Setter
public class CoachingPlan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(optional = false, fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String goal;

    @Convert(converter = StringListJsonConverter.class)
    @Column(name = "principles_json", nullable = false, columnDefinition = "text")
    private List<String> principles;

    @Convert(converter = StringListJsonConverter.class)
    @Column(name = "priorities_json", nullable = false, columnDefinition = "text")
    private List<String> priorities;

    @Convert(converter = StringListJsonConverter.class)
    @Column(name = "actions_json", nullable = false, columnDefinition = "text")
    private List<String> actions;

    @Column(name = "start_date", nullable = false)
    private LocalDate startDate;

    @Column(name = "review_date")
    private LocalDate reviewDate;

    @Column(columnDefinition = "text")
    private String notes;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}
